fetcher_test.go

```diff
@@ -25,6 +30,63 @@ func TestNewClientCredentialsFlowExpectsObjectInitializedSuccessfully(t *testing
 	assert.Equal(t, grantType, a.grantType, "expected grant type to match")
 }
 
+func TestConfirmsTokenNotCaching(t *testing.T) {
+	i := 1
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintln(w, "{\"access_token\":\"token"+strconv.Itoa(i)+"\"}")
+		i++
+	}))
+	defer ts.Close()
+
+	// arrange
+	clientID := "PV2AvGcMjOFErV6QpaqKnfrUdt8yPuHI"
+	clientSecret := "9oXvXvWHfQaaAiWr-wBfS5Vtyp3aGyMuwIwqYs2NuRtmV7-1XEXXNJ1ZA97jLo6J"
+
+	// act
+	f := NewTokenFetcher(ts.Client(), ts.URL, clientID, clientSecret)
+	token, err := f.CachedToken("audience")
+	token2, err := f.Token("audience")
```

Comment:
should this be CachedToken? not clear from test name that it is testing that .Token does not return same return as .CachedToken? not quite clear what the intent of this test is